lib/pbench/server/api/resources/datasets_list.py

```diff
@@ -572,13 +575,17 @@ def keyspace(self, query: Query) -> JSONOBJECT:
         Database.dump_query(query, current_app.logger)
         aggregate: JSONOBJECT = {}
 
-        datasets = query.all()
-        for d in datasets:
+        results = query.all()
+        for result in results:
+            # NOTE: to allow sorting by User.username, our query is defined to
+            # return (Dataset, User), so we need to isolate the Dataset from
+            # the tuple.
+            dataset = result[0]
```

Comment:
You could make this code self-documenting as follows:
```python
            dataset, _user = result
```
and then you wouldn't need the code comment.  (You could use just `_` instead of `_user` if you don't want the extra context for the reader).

Ditto line 710.

Reply:
The comment explains the *why* rather than just implying the structure of the `tuple`, and you know I hate underscored names. Yeah, `dataset, _` would work here and isn't quite as ugly, but lone underscores are easy to miss.